api/lib/domain/services/verify-certificate-code-service.js

```diff
@@ -0,0 +1,22 @@
+const _ = require('lodash');
+const certificationCourseRepository = require('../../../lib/infrastructure/repositories/certification-course-repository');
+
+const availableCharacters = '2346789BCDFGHJKMPQRTVWXY'.split('');
+const NB_CHAR = 6;
+
+function _generateCode() {
+  return 'P-' + _.times(NB_CHAR, _randomCharacter).join('');
+}
+
+function _randomCharacter() {
+  return _.sample(availableCharacters);
+}
+
+module.exports = {
+
+  async generateCertificateVerificationCode(generateCode = _generateCode) {
```

Comment:
hmm marrant ça, c'est la première fois que je vois ça. C'est plus pratique pour les tests ? Ou bien c'est une pensée de prévision de réusabilité par une autre fonction de génération ?

Reply:
Oui c'est une façon de pouvoir tester le service sans stubber `_.sample` qui ne nous appartient pas (_Don't mock what you don't own_)